drivers/drivers_initialize.c

```diff
@@ -49,6 +49,15 @@
 #include <nuttx/virtio/virtio.h>
 #include <nuttx/drivers/optee.h>
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if (defined(CONFIG_LWL_CONSOLE) + defined(CONFIG_CONSOLE_SYSLOG) +     \
+     defined(CONFIG_SERIAL_RTT_CONSOLE) + defined(CONFIG_RPMSG_UART_CONSOLE)) > 1
+#  error More than one console driver selected. Check your configuration !
```

Comment:
Is it possible to include which ones are included? Maybe:
```
#define MULTCONSOLES do { defined(CONFIG_LWL_CONSOLE) ? "LWL_CONSOLE, " : defined(CONFIG_CONSOLE_SYSLOG) ? "CONSOLE_SYSLOG, " :  defined(CONFIG_SERIAL_RTT_CONSOLE) ? "RTT_CONSOLE, " : defined(CONFIG_RPMSG_UART_CONSOLE) ? "RPMSG_UART_CONSOLE. " : ". " } while(0);

#  error "More than one console driver selected: " MULTCONSOLES "Check your configuration !"
```

Not sure if this approach will work, but I think it will.

Reply:
#error doesn't support it, the only solution is using _Static_assert(), but it is a C11 feature:
https://stackoverflow.com/questions/57608296/error-directive-in-c-how-to-display-some-define

Reply:
I don't know if it makes sense to complicate this check. Verifying which CONSOLE options are enabled is only a matter of searching `.config` for the string "_CONSOLE".

Reply:
Yes, it is easy to find. But for a beginner it "More than one console driver selected" could be not very informative. The idea of searching "_CONSOLE" could help!

Reply:
how about move ALL CONFIG_XXX_CONSOLE  under a choice.

Reply:
> how about move ALL CONFIG_XXX_CONSOLE under a choice?

That would be ideal, but at the moment it's not possible. Some console drivers depends on `CONFIG_XXX_CONSOLE` options but other use Kconfig `select CONFIG_XXX_CONSOLE` to mark that this device provides console. There is no consistency between drivers so fixing this is not that easy.